### DISCUSSION TITLE
ABP Permission Management supports caching?

### DISCUSSION THREAD

**Discussion by kbsiva** [abpframework/abp#9065] [Q&A]
Hi Team,
 I have a question. I have various permission providers and will ABP Permission management supports caching mechanism?
If so please can you provide any information on that.

Thanks
Siva Kumar KB

**Comment by realLiangshiwei** [comment]
Permission management use `IDistributedCache`

https://github.com/abpframework/abp/blob/67932648b52a2f069f23a4ba3e46ad749f0db9d1/modules/permission-management/src/Volo.Abp.PermissionManagement.Domain/Volo/Abp/PermissionManagement/PermissionStore.cs#L20

**Reply by kbsiva** [reply]
So, If I use Permission management it's already handled cache right? Do I need to do any specific extension to the method or configure any part of the code to see that caching is enabled and for testing purpose?

**Reply by realLiangshiwei** [reply]
It works right out of the box
